core/src/test/java/org/keycloak/sdjwt/sdjwtvp/SdJwtVPTest.java

```diff
@@ -196,4 +198,24 @@ public void testS6_2_PresentationPartialDisclosure() throws VerificationExceptio
                 .verifySignature(TestSettings.verifierContextFrom(presenteSdJwtVP.getCnfClaim(), "ES256"));
     }
 
+
+    @Test
+    public void testOf_validInput() {
+        String sdJwtString = TestUtils.readFileAsString(getClass(), "sdjwt/s6.2-presented-sdjwtvp.txt");
+        SdJwtVP sdJwtVP = SdJwtVP.of(sdJwtString);
+
+        assertNotNull(sdJwtVP);
+        assertEquals(4, sdJwtVP.getDisclosures().size());
+    }
+
+    @Test
+    public void testOf_MalformedSdJwt_ThrowsIllegalArgumentException() {
+        // Given
+        String malformedSdJwt = "issuer-signed-jwt"; // missing delimiter at the end
+
+        // When & Then
+        var exception = assertThrows(IllegalArgumentException.class, () -> SdJwtVP.of(malformedSdJwt));
+        assertEquals("SD-JWT is malformed, expected to end with ~", exception.getMessage());
+    }
```

Comment:
This test will likely fail as is due to the asserted error message not aligning with the changes. Additionally, the test vector should have a different comment, since as highlighted by  @ahus1, the problem is not that the SD-JWT is not ending with ~.

Reply:
@IngridPuppet Thank you for pointing out the issue with the test. I'll update the test to reflect the changes made to the error message. I'll make sure to address this in the next update.